ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                    poFeatureDefn->SetGeomType(wkbPoint);
+            }
+            else if (phMiraMonLayer->bIsArc && !phMiraMonLayer->bIsPolygon)
+            {
+                if (phMiraMonLayer->TopHeader.bIs3d)
+                    poFeatureDefn->SetGeomType(wkbLineString25D);
+                else
+                    poFeatureDefn->SetGeomType(wkbLineString);
+            }
+            else if (phMiraMonLayer->bIsPolygon)
+            {
+                // 3D
+                if (phMiraMonLayer->TopHeader.bIs3d)
+                {
+                    if (phMiraMonLayer->TopHeader.bIsMultipolygon)
+                        poFeatureDefn->SetGeomType(wkbMultiPolygon25D);
+                    else
+                        poFeatureDefn->SetGeomType(wkbPolygon25D);
+                }
+                else
+                {
+                    if (phMiraMonLayer->TopHeader.bIsMultipolygon)
+                        poFeatureDefn->SetGeomType(wkbMultiPolygon);
+                    else
+                        poFeatureDefn->SetGeomType(wkbPolygon);
+                }
+            }
+            else
+            {
+                CPLError(CE_Failure, CPLE_NotSupported,
+                         "MiraMon file type not supported.");
+                bValidFile = false;
+                return;
+            }
+
+            if (phMiraMonLayer->TopHeader.bIs3d)
+            {
+                const char *szHeight =
+                    CSLFetchNameValue(papszOpenOptions, "Height");
+                if (szHeight)
+                {
+                    if (!strcasecmp(szHeight, "Highest"))
+                        phMiraMonLayer->nSelectCoordz =
+                            MM_SELECT_HIGHEST_COORDZ;
+                    else if (!strcasecmp(szHeight, "Lowest"))
+                        phMiraMonLayer->nSelectCoordz = MM_SELECT_LOWEST_COORDZ;
+                    else
+                        phMiraMonLayer->nSelectCoordz = MM_SELECT_FIRST_COORDZ;
+                }
+                else
+                    phMiraMonLayer->nSelectCoordz = MM_SELECT_FIRST_COORDZ;
+            }
+
+            /* ------------------------------------------------------------ */
+            /*      Establish the nMemoryRatio to use                       */
+            /* ------------------------------------------------------------ */
+            const char *pszMemoryRatio =
+                CSLFetchNameValue(papszOpenOptions, "OpenMemoryRatio");
+
+            if (pszMemoryRatio)
+                phMiraMonLayer->nMemoryRatio = atof(pszMemoryRatio);
+            else
+                phMiraMonLayer->nMemoryRatio = 1;  // Default
+
+            if (phMiraMonLayer->nMemoryRatio < 0)
+                phMiraMonLayer->nMemoryRatio =
+                    fabs(phMiraMonLayer->nMemoryRatio);
+
+            // It has no sense try to work with less ratio than 0.25.
+            if (phMiraMonLayer->nMemoryRatio < 0.25)
+                phMiraMonLayer->nMemoryRatio = 0.25;
```

Comment:
so perhaps the following ?  (bounds likely to be set consistent with the update case above)
```suggestion
            if (pszMemoryRatio)
                phMiraMonLayer->nMemoryRatio = std::clamp(CPLAtof(pszMemoryRatio), 0.25, 2);
            else
                phMiraMonLayer->nMemoryRatio = 1;  // Default
```

Reply:
Yes, done!